dataflow/gpu-examples/tensorflow-landsat/main.py

```diff
@@ -249,33 +255,34 @@ def run(
         vis_params: Visualization parameters including {rgb_bands, min, max, gamma}.
         beam_args: Optional list of arguments for Beam pipeline options.
     """
-    rgb_band_names = vis_params["rgb_band_names"]
-    min_value = vis_params["min"]
-    max_value = vis_params["max"]
-    gamma = vis_params["gamma"]
+    rgb_band_names = vis_params.get("rgb_band_names", DEFAULT_RGB_BAND_NAMES)
+    min_value = vis_params.get("min", DEFAULT_MIN_BAND_VALUE)
+    max_value = vis_params.get("max", DEFAULT_MAX_BAND_VALUE)
+    gamma = vis_params.get("gamma", DEFAULT_GAMMA)
 
+    gpu_hint = f"type:{gpu_type};count:{gpu_count};install-nvidia-driver"
     beam_options = PipelineOptions(beam_args, save_main_session=True)
     pipeline = beam.Pipeline(options=beam_options)
     (
         pipeline
         | "Create scene IDs" >> beam.Create(scenes)
-        | "Check GPU availability"
-        >> beam.Map(
+        | "Check GPU availability" >> beam.Map(
             lambda x, unused_side_input: x,
             unused_side_input=beam.pvalue.AsSingleton(
-                pipeline | beam.Create([None]) | beam.Map(check_gpus)
+                pipeline
+                | beam.Create([None])
+                | beam.Map(check_gpus).with_resource_hints(accelerator=gpu_hint)
             ),
         )
         | "Get RGB band paths" >> beam.Map(get_band_paths, rgb_band_names)
+        | "Reshuffle" >> beam.Reshuffle()
```

Comment:
Explain with comments why we are using Reshuffle, can link the break fusion page in Dataflow docs

Reply:
Thanks, adding a comment and linking to https://cloud.google.com/dataflow/docs/guides/deploying-a-pipeline#preventing-fusion